tests/kernel/lifo/lifo_usage/src/main.c

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <ztest.h>
+#include "lifo_usage.h"
+#include <kernel.h>
+
+#define STACK_SIZE 1024
+#define LIST_LEN 2
+
+struct k_lifo lifo, plifo;
+static ldata_t data[LIST_LEN];
+struct k_fifo timeout_order_lifo;
+
+static struct k_thread tdata, tdata1;
+static K_THREAD_STACK_DEFINE(tstack, STACK_SIZE);
+static K_THREAD_STACK_DEFINE(tstack1, STACK_SIZE);
+
+static struct k_sem start_sema, wait_sema;
+
+struct scratch_lifo_packet {
+	void *link_in_lifo;
+	void *data_if_needed;
+};
+
+struct reply_packet {
+	void *link_in_fifo;
+	s32_t reply;
+};
+
+struct timeout_order_data {
+	void *link_in_lifo;
+	struct k_lifo *klifo;
+	u32_t timeout;
+	s32_t timeout_order;
+	s32_t q_order;
+};
+
+static struct k_lifo lifo_timeout[2];
+
+struct timeout_order_data timeout_order_data[] = {
+	{0, &lifo_timeout[0], 20, 2, 0},
```

Comment:
I am expecting 20 milisecond here. So I can use MSEC(x). However if I use MSEC(20) then it gives 2 milisecond instead of 20. There is dependency on sys_clock_ticks_per_sec. So I can't use it